components/common/src/types.rs

```diff
@@ -390,6 +390,30 @@ impl ListenCtlAddr {
     pub fn ip(&self) -> IpAddr { self.0.ip() }
 
     pub fn port(&self) -> u16 { self.0.port() }
+
+    // TODO (CM): This really should be FromStr, but we can't use
+    // FromStr, because the current implementation of
+    // env_config_socketaddr! (and ultimately env_config!) macro
+    // defines one for us.
+    pub fn resolve_listen_ctl_addr(input: &str) -> crate::error::Result<ListenCtlAddr> {
+        let listen_ctl_addr = if input.find(':').is_some() {
```

Comment:
We have very similar logic in two other places ([1](https://github.com/habitat-sh/habitat/blob/77b980bd97aff98f0702ba6845b10e8b6283d6ad/components/hab/src/cli/hab/util.rs#L125), [2](https://github.com/habitat-sh/habitat/blob/77b980bd97aff98f0702ba6845b10e8b6283d6ad/components/sup/src/manager/peer_watcher.rs#L112)). It is slightly different because this methods tries to actually resolve the address. Maybe the other instances should too though? This is just an observation I dont think anything is necessary for this PR.

Reply:
Good find... yeah, we should clean that up. I'll take a look at that for a follow-on PR :+1: